app/src/main/java/com/mapbox/navigation/examples/basics/MultipleWayPointsActivity.kt

```diff
@@ -0,0 +1,557 @@
+        mapboxNavigation.registerVoiceInstructionsObserver(voiceInstructionsObserver)
+
+        if (mapboxNavigation.getRoutes().isEmpty()) {
+            // if simulation is enabled (ReplayLocationEngine set to NavigationOptions)
+            // but we're not simulating yet,
+            // push a single location sample to establish origin
+            mapboxReplayer.pushEvents(
+                listOf(
+                    ReplayRouteMapper.mapToUpdateLocation(
+                        eventTimestamp = 0.0,
+                        point = Point.fromLngLat(-122.39726512303575, 37.785128345296805)
+                    )
+                )
+            )
+            mapboxReplayer.playFirstLocation()
+        }
+    }
+
+    override fun onBackPressed() {
+        // clear current route
+        if (mapboxNavigation.getRoutes().isNotEmpty()) {
+            stopSimulation()
+            mapboxNavigation.setRoutes(emptyList())
+            addedWaypoints.clear()
+        } else {
+            super.onBackPressed()
+        }
+    }
+
+    override fun onStop() {
+        super.onStop()
+        // unregister event listeners to prevent leaks or unnecessary resource consumption
+        mapboxNavigation.unregisterRoutesObserver(routesObserver)
+        mapboxNavigation.unregisterLocationObserver(locationObserver)
+        mapboxNavigation.unregisterRouteProgressObserver(routeProgressObserver)
+        mapboxNavigation.unregisterVoiceInstructionsObserver(voiceInstructionsObserver)
+    }
+
+    override fun onDestroy() {
+        super.onDestroy()
+        MapboxNavigationProvider.destroy()
+        speechApi.cancel()
+        voiceInstructionsPlayer.shutdown()
+    }
+
+    private fun addWaypoint(destination: Point) {
+        val originLocation = navigationLocationProvider.lastLocation
+        val originPoint = originLocation?.let {
+            Point.fromLngLat(it.longitude, it.latitude)
+        } ?: return
+
+        if (addedWaypoints.isEmpty) {
+            addedWaypoints.addRegular(originPoint)
+        }
+
+        when (val index = binding.waypointTypeView.selectedTabPosition) {
+            1 -> addedWaypoints.addNamed(destination, "Example of the named waypoint")
+            0 -> addedWaypoints.addRegular(destination)
+            2 -> addedWaypoints.addSilent(destination)
+            else -> error("Not supported type of waypoint. index $index")
+        }
+
+        // execute a route request
+        // it's recommended to use the
+        // applyDefaultNavigationOptions and applyLanguageAndVoiceUnitOptions
+        // that make sure the route request is optimized
+        // to allow for support of all of the Navigation SDK features
+        mapboxNavigation.requestRoutes(
+            RouteOptions.builder()
+                .applyDefaultNavigationOptions()
+                .applyLanguageAndVoiceUnitOptions(this)
+                .coordinatesList(addedWaypoints.coordinatesList())
```

Comment:
This example focuses only on multiple waypoint. Let's keep it simple and let our users learn features one by one. Anyway if we find out later that the layers and the bearing is very very important and users don't realise their importance because of misleading examples - let's make the examples more complex then.
What do you think about it?

Reply:
> Silent waypoints do not become actually **silent**
> 
>  device-2021-11-09-144852.mp4

That's a very good point! Silent waypoints become silent when there is at least a one regular waypoint after it. See the [video](https://user-images.githubusercontent.com/6190346/140930194-e8dade27-f718-4314-adaf-76d99278ed46.mp4)

I'm not sure how to handle the case. I see two options.
1. We can leave example as is. It's simple and demonstrates how the SDK handles different waypoints.
2. I can add following logic: If the last waypoint is silent convert it to a regular one. This will complicate the example but will make the behaviour more obvious.

@RingerJK , what would you recommend me to do?

Reply:
@RingerJK , I also don't remember any documentation about this behaviour. Could it be changed? I don't know 🤔

Reply:
I vote for the second option. We should provide reliable examples, even if customers would like to copy-paste it to the app - it has to work